packages/jaeger-ui/src/components/SearchTracePage/SearchResults/ResultItem.test.js

```diff
@@ -13,33 +13,97 @@
 // limitations under the License.
 
 import React from 'react';
-import { Tag } from 'antd';
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { MemoryRouter } from 'react-router-dom';
 
 import ResultItem from './ResultItem';
 import * as markers from './ResultItem.markers';
 import traceGenerator from '../../../demo/trace-generators';
 import transformTraceData from '../../../model/transform-trace-data';
 
-const trace = transformTraceData(traceGenerator.trace({}));
+// Helper function to wrap component with Router
+const renderWithRouter = ui => {
+  return render(ui, { wrapper: MemoryRouter });
+};
+
+let trace; // Use let to allow modification in tests
+
+beforeEach(() => {
+  // Reset trace data before each test
```

Comment:
Why do we need to reset it for each test? Is something modifying the data?

Reply:
The test case that modifies the trace data is:
`it('<ResultItem /> should render error icon on ServiceTags that have an error tag'`
This test directly adds an error tag to one of the spans within the trace object.